code/__HELPERS/priority_announce.dm

```diff
@@ -36,8 +36,9 @@
  * * players - a list of all players to send the message to. defaults to all players (not including new players)
  * * encode_title - if TRUE, the title will be HTML encoded
  * * encode_text - if TRUE, the text will be HTML encoded
+ * * send_to_world - if TRUE will send the announcement to the world rather than a list of players
  */
-/proc/priority_announce(text, title = "", sound, type, sender_override, has_important_message = FALSE, list/mob/players = GLOB.player_list, encode_title = TRUE, encode_text = TRUE, color_override)
+/proc/priority_announce(text, title = "", sound, type, sender_override, has_important_message = FALSE, list/mob/players = GLOB.player_list, encode_title = TRUE, encode_text = TRUE, color_override, send_to_world = FALSE)
```

Comment:
I don't like this parameter, and I don't necessarily like using this proc for out-of-ic announcements

Case in point: In your current implementation, you're sending admin announces to newscasters!

If you want the formatting it offers specifically, we should separate the formatting part into a separate proc.

Reply:
I was thinking whether to split it or not, so yeah I'll look into that.

Reply:
can we just check for players being set to world

Reply:
`players = world` is an option but I would still prefer this proc stay for IC stuff so we don't accidentally leak stuff to newscasters (it's already happened once and almost happened twice)